benchmark/src/libfork_benchmark/uts/external/uts.c

```
@@ -0,0 +1,474 @@
+
+  // given target b_i, find prob p so expected value of
+  // geometric distribution is b_i.
+  p = 1.0 / (1.0 + b_i);
+
+  // get uniform random number on [0,1)
+  h = rng_rand(parent->state.state);
+  u = rng_toProb(h);
+
+  // max number of children at this cumulative probability
+  // (from inverse geometric cumulative density function)
+  numChildren = (int)floor(log(1 - u) / log(1 - p));
```

Cppcheck (2.20.0)

[warning] Invalid log() argument nr 1. The value is 0 but the valid values are '4.94066e-324

(invalidFunctionArg)